File: PythonLibs/forestBrain.py
# forestBrain.py - Uses scikit-learn RandomForest
from sklearn.datasets import load_wine
from sklearn.ensemble import RandomForestClassifier

def train_forest():
    data = load_wine()
    rf = RandomForestClassifier(n_estimators=10)
    rf.fit(data.data, data.target)
    print('RandomForestClassifier trained on Wine dataset')

if __name__ == '__main__':
    train_forest()

File: C++Libs/forestBrain.cpp
// forestBrain.cpp - Uses mlpack Random Forest
#include <mlpack/core.hpp>
#include <mlpack/methods/random_forest/random_forest.hpp>
#include <armadillo>
#include <iostream>

using namespace mlpack;
using namespace mlpack::ensemble;
using namespace arma;

int main() {
    mat data;
    data::Load("wine.csv", data, true);
    Row<size_t> labels = conv_to<Row<size_t>>::from(data.row(data.n_rows-1));
    data.shed_row(data.n_rows-1);
    RandomForest<> rf(data, labels, 3, 10);
    std::cout << "RandomForest trained on Wine dataset" << std::endl;
    return 0;
}
